src/Todo/Model.hs

```diff
@@ -31,4 +49,32 @@ data Todo = Todo
     , todoSubtasks    :: Todos
     , todoDeadline    :: Maybe Deadline
     , todoDescription :: Maybe String
-    } deriving (Eq, Show)
+    } deriving (Eq, Show, Generic)
+
+instance Hashable Todo 
+
+defTodo :: Title -> Priority -> Tags -> Todo
+defTodo todoTitle todoPriority todoTags = Todo { todoState = TODO
+                                               , todoSubtasks = HSet.empty
+                                               , todoDeadline = Nothing
+                                               , todoDescription = Nothing
+                                               , .. }
+
+getByState :: State -> Todos -> Todos
+getByState state = HSet.filter $ (state ==) . todoState
+
+getByTag :: Tag -> Todos -> Todos
+getByTag tag = HSet.filter $ NESet.member tag . todoTags
+
+getTopPrio :: Int -> Todos -> Todos
+getTopPrio n = HSet.fromList . take n . sortBy (flip $ comparing todoPriority) . toList
+
+tagStatistics :: Todos -> HashMap Tag Int
+tagStatistics = count . concatMap (toList . todoTags)
+  where
+    count :: (Eq a, Hashable a) => [a] -> HashMap a Int
+    count [] = mempty
+    count (x:xs) = HMap.insert x (cnt + 1) mapp
```

Comment:
Okay, prepare for `unordered-containers` kung-foo. This library actually has some useful functions.

If you want to insert key with value 1 if it's first key and add 1 to existing value if this is key is already in HashMap you can use [`insertWith`](https://hackage.haskell.org/package/unordered-containers-0.2.9.0/docs/Data-HashMap-Lazy.html#v:insertWith) function.

And you don't need to write pattern-matching on list. You can just use `foldr` function. Unfortuately, your implementation and implementation with `foldr` function is/won't be efficient and tail-recursive. We didn't discuss efficiency in Haskell code yet. But it's better use [`fodl'`](https://hackage.haskell.org/package/base-4.11.0.0/docs/Data-List.html#v:foldl-39-) function.

So, basically, just implement `count` using `foldl'` and `insertWith` functions.